### ISSUE TITLE
Feature Request: New Yale Home app.  Will it break anything?

### ISSUE THREAD

**Issue by networkoctopus** [homebridge-plugins/homebridge-august#60]
### Problem

More an awareness thing.  As of today, Yale are transitioning away from Yale Access app, to Yale home app. Not sure if this will break homebridge support with this plugin or not.

### Solution

Maybe as simple as adding updated links in code to new Yale domains? 

### Alternatives

To login via web, my app pushes me to https://account.aaecosystem.com/?brand=yale , whereas previously I'd used account.august.com to register the lock and get the verification pin.

### Additional context

I have migrated my account to new Yale home, but not yet home to confirm if its broken the plugin.  Thanks Donavan.

**Comment by donavanbecker** [comment]
Looking at the website, this like it is only Asia Pacfic

**Comment by kurtern84** [comment]
Can confirm it is broken after the update.

**Comment by AquaRelliux** [comment]
Yes it just broke for me also. Any way to fix this? once the user account is migrated it can't be logged into in August :(

**Comment by donavanbecker** [comment]
I would first open an issue on the august api repo that I use: [august-api](https://github.com/hufftheweevil/august-api)

**Comment by AquaRelliux** [comment]
@donavanbecker Thank you but I doubt it will fix this issue. The reason it breaks is because the users accounts as of now no longer exists on the august database. Yale has decided to migrate them to a new infrastructure which means this plugin unless changed will not work for Yale door locks any longer. Only way to fix it I guess is to use their new portal which has been posted above I am not sure if this API can be used with Yale in that case

**Comment by AquaRelliux** [comment]
Looking at it it seems to be an API avaiable for Yale door locks here:

https://docs.seam.co/latest/device-guides/get-started-with-yale-locks

Maybe this is something you could integrate in this plugin @donavanbecker I would love if you could :)

**Comment by donavanbecker** [comment]
> @donavanbecker Thank you but I doubt it will fix this issue. The reason it breaks is because the users accounts as of now no longer exists on the august database. Yale has decided to migrate them to a new infrastructure which means this plugin unless changed will not work for Yale door locks any longer. Only way to fix it I guess is to use their new portal which has been posted above I am not sure if this API can be used with Yale in that case

https://github.com/bdraco/yalexs/pull/61

**Comment by rubenstolk** [comment]
Hmm I guess I shouldn't have upgraded yet....

**Comment by networkoctopus** [comment]
Yale only giving folks until start of July to migrate before losing access
natively even, so we don’t have much choice!

**Comment by AquaRelliux** [comment]
Any updates on this? real bummer this does not work anymore to be honest one of the best plugins for Homebridge

**Comment by donavanbecker** [comment]
No one has open an issue on august-api so not much that I can do with the plugin

**Comment by AquaRelliux** [comment]
Not so technical but the api I posted above can't that accomplish the same thing? It seems to be updated and uses the same login method as the August one. This should not be on the August api since the users has been migrated away from that platform.

**Comment by donavanbecker** [comment]
Sorry, I can look into it, but don't have time right now.

**Comment by AquaRelliux** [comment]
> Sorry, I can look into it, but don't have time right now.

That is understandable! I will send some contribution your way if you have time to fix this :) API I linked looked pretty straight forward and you can just make a switch in the plugin that switches if it is a August lock or Yale lock device.

**Comment by AquaRelliux** [comment]
This seems to have been fixed on the Homeassistant plugin for August now. Can we just implement the same changes they did on this plugin?

**Comment by donavanbecker** [comment]
@AquaRelliux Pull requests are welcomed.

**Comment by AquaRelliux** [comment]
> @AquaRelliux Pull requests are welcomed.

Unfortunately I don't know how to code this at all believe me I would give it an honest attempt if I knew where to even begin I think it is just to much for a beginner on this to understand :)

**Comment by AquaRelliux** [comment]
So I migrated this to Homeassistant August plugin it works great with homekit if anyone else also has this problem with August. The plugin on Homeassistant has now builtin support got Yale access and Yale home as well as the original August auth. I hope this gets fixed on Homebridge soon since I love the simplicity over Homeassistant but seems like developers on Homeassistant have more resources to help out which is a shame.

**Comment by AquaRelliux** [comment]
@donavanbecker any update on a fix for this?

**Comment by donavanbecker** [comment]
Sorry it’s summertime, so not much time to work on this right now

**Comment by donavanbecker** [comment]
v1.2.0 has been released, but won't fix this when I get time I will look at adding this support, might be in a different plugin though.

**Comment by simonrb2000** [comment]
A new plugin? Interesting 😉

**Comment by OliveFinch** [comment]
I've also had to disable this plugin as I've migrated to the new Yale account. I'm in the UK so I suppose that all European customers will have this issue also.

**Comment by Antonk123** [comment]
Anything new on this problem? :)

**Comment by peedub** [comment]
The plugin for homekit updated a couple of months ago to accommodate this - is there potentially something there that could help with this issue?

**Comment by donavanbecker** [comment]
Unfortunately not going to happen anytime soon. Don't see any promising apis that make it easy to setup.

**Comment by AquaRelliux** [comment]
> Unfortunately not going to happen anytime soon. Don't see any promising apis that make it easy to setup.

How is it that it works in Homeassistant then? They also have a August plugin and it was fixed within days after it broke? They must use an API?

**Comment by donavanbecker** [comment]
@AquaRelliux this integrates with a module that manages all of the API components. So I don't really deal with much with the API.

If that module was updated to support Yale Home then I could add it to the plugin. Or any module that supports it.

**Comment by donavanbecker** [comment]
Also it is hard to test this without an account.

**Comment by donavanbecker** [comment]
Okay, I may have found something which I have added into the latest beta. Without an account though, I can't test it. 

Please test the latest `v1.3.0` beta and report back if it works for you.

you will have to set a new config: `countryCode`, anything besides `US` will force to use Yale Home url.

<a><img src="https://github.com/donavanbecker/homebridge-august/assets/9875439/00bff58e-93ea-45b7-9088-e169d7ae15a8" width="500px"></a>

**Comment by simonrb2000** [comment]
Nearly! I got the August code and it finds my 3 door locks however it then throws a fault.

Hope the screenshot helps, I can grab proper logs if that helps.

The plugin used to work prior to the account change and since then it’s been disabled.

[image0.png]

Sent from my iPhone

On 29 Oct 2023, at 03:25, Donavan Becker ***@***.***> wrote:

﻿

Okay, I may have found something which I have added into the latest beta. Without an account though, I can't test it.

Please test the latest v1.3.0 beta and report back if it works for you.

**Comment by simonrb2000** [comment]
![IMG_0610](https://github.com/donavanbecker/homebridge-august/assets/15991296/0152be9d-bd64-402b-bbb3-1a14a65aa88e)

**Comment by donavanbecker** [comment]
With the august-api now support the YaleHome URL, I have Release v1.3.0. I am going to close this issue. If you are having issues with v1.3.0 please open a new issue.

**Comment by donavanbecker** [comment]
Follow #75

**Comment by AquaRelliux** [comment]
I tried this today I get this error:
 
![image](https://github.com/donavanbecker/homebridge-august/assets/15650734/cf52740e-d9f5-401e-98e4-ceac9df69853)

So no still does not work

**Comment by donavanbecker** [comment]
> Follow #75

We have to wait for an API fix. So follow #75

**Comment by AquaRelliux** [comment]
I dont understand. How can this be an API problem when it has been working in Homeassistant for +1 year now since this problem was first discovered? Homeassistant fixed this issue within a week. Do they use a different API?

**Comment by donavanbecker** [comment]
See: https://github.com/donavanbecker/homebridge-august/issues/75#issuecomment-1858398207

**Comment by donavanbecker** [comment]
If someone wants to create a new module that connects to the API, I would definitely do that. I don't know the technicalities behind the module (API) so have to rely on that module.

I found that module when I was first wanting to connect a lock that I had that wasn't able to connect to HomeKit. So I made this plugin, but I have to rely on that module to be updated to be able too update this plugin.
